salt/journal-cms/init.sls

```diff
@@ -91,6 +91,17 @@ composer-install:
             - install-composer
             - journal-cms-localhost
 
+# these files accumulate over time and are not required in non-prod environments
+{% if pillar.elife.env not in ['prod'] %}
+prune-accumulating-files:
+    cmd.run:
+        - name: rm -rf /srv/journal-cms/web/sites/default/files
+        - require:
+            - journal-cms-repository
+        - require_in:
+            - cmd: web-sites-file-permissions
+{% endif %}
```

Comment:
It would be my preference that this step was only performed on the same frequency as `site-install`. I will have a go at the refactor.

Reply:
Something like this:

```
# these files accumulate over time and are not required in non-prod environments
prune-accumulating-files:
    cmd.run:
        - name: rm -rf /srv/journal-cms/web/sites/default/files
        - require:
            - journal-cms-repository
        - require_in:
            - cmd: site-install
        - unless:
            - sudo -u {{ pillar.elife.deploy_user.username}} ../vendor/bin/drush cget system.site name
```

Empty the folder prior to `site-install` if we detect that drupal is not installed.

Reply:
This would mean that directory is effectively never deleted in the environments that need it.

1. run this *before* `site-install` is called
2. only run if the site isn't installed
3. in `dev` and `ci` environments, `site-install` is *always* called

Given these conditions, whenever the 'prune' state is reached it's always seeing an installed site *except* for the first time the instance is built.

> It would be my preference that this step was only performed on the same frequency as site-install. 

Could I know why? It might help me unjumble this logic a bit better. Is it speed reasons?

Reply:
I believe this is only an issue on instances where drupal is reinstalled repeatedly rendering all the files in the files folder orphaned. On a cleanup of journal cms end2end which we are performing daily we should be reinstalling drupal. For instances such as continuumtest it may be valuable to ensure that files upload in the CMS survive longer than every time `sudo salt-call state.highstate` is called.

Reply:
The code I submitted was just a suggestion and I haven't tested it out.

Reply:
>  On a cleanup of journal cms end2end which we are performing daily we should be reinstalling drupal. 

Actually, end2end is turned off most of the time. There is a ['rollback' step in the journal-cms Jenkinsfile](https://github.com/elifesciences/journal-cms/blob/develop/Jenkinsfile#L22-L25) that looks like it resets the end2end environment back to the last 'approved' commit on failure.

If I'm re-reading the comment history correctly then:

* `ci` should get this
* `end2end` should get this
* `continuumtest` should not
* `prod` absolutely should not

Your code suggestion (thankyou) has a double negative which I might avoid but if the targeted environments are correct I can go from there.